### ISSUE TITLE
Attempting to install template results in `ERROR: qrexec call 'qubes.TemplateSearch' failed: unexpected data format.`

### ISSUE THREAD

**Issue by andrewdavidwong** [QubesOS/qubes-issues#8303]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release

4.1

### Brief summary

Attempting to install a template results in:

```
ERROR: qrexec call 'qubes.TemplateSearch' failed: unexpected data format.
```

### Steps to reproduce

1. Create an update qube based on `debian-11-minimal` with required packages (including [`qubes-repo-templates`](https://github.com/QubesOS/qubes-issues/issues/7605)).
2. Set dom0 updates to use the update qube created in step 1.
3. In dom0, execute:
   ```
   qvm-template install fedora-38
   ```

### Expected behavior

`fedora-38` template installs.

### Actual behavior

```
ERROR: qrexec call 'qubes.TemplateSearch' failed: unexpected data format.
```

Same result when trying to install `debian-11`.

**Comment by adrelanos** [comment]
I am also running into this. Please re-open.

This needs a better error message.

User report:
https://forum.qubes-os.org/t/qubes-users-template-manager-fails-to-fetch-template-list-after-in-place-upgrade-to-4-2/26355

Potentially related:
https://github.com/QubesOS/qubes-issues/issues/7605

`qubes-repo-templates` and `qubes-core-agent-dom0-updates` are being installed in dom0 (`qubes-prefs updatevm`) `sys-whonix`.

    dpkg -l | grep qubes-repo-templates

> ii  qubes-repo-templates                     4.3.2-1+deb13u1                      amd64        Repository definition for Qubes OS VM template packages.

    dpkg -l | grep dom0-update

> ii  qubes-core-agent-dom0-updates            4.3.37-1+deb13u1                     amd64        Scripts required to handle dom0 updates.

    find /etc/qubes/repo-templates

```
/etc/qubes/repo-templates
/etc/qubes/repo-templates/qubes-templates.repo
/etc/qubes/repo-templates/keys
/etc/qubes/repo-templates/keys/RPM-GPG-KEY-qubes-4.3-primary
/etc/qubes/repo-templates/keys/RPM-GPG-KEY-qubes-4.3-templates-community
```

No related configuration files are modified.

    sudo debsums -ce

```
/etc/machine-id
/etc/dnf/dnf.conf
/etc/apt/sources.list.d/qubes-r4.list
/etc/tinyproxy/tinyproxy-updates.conf
```

**Comment by adrelanos** [comment]
Diagnosed.

> `qubes.TemplateSearch qrexec action is prohibited in PERSISTENT Mode - SYSMAINT Session.`

It's a bit difficult to diagnose because neither dom0 nor VM journal has any output pointing. I had to modify dom0 `qvm-template` command to get debug output.

**Comment by ben-grande** [comment]
@adrelanos Fixed in latest Whonix 18?

**Comment by adrelanos** [comment]
Yes. A better error message has been implemented in file `/usr/libexec/user-sysmaint-split/qubes-rpc-block`. Will look similar to this:

> Qrexec action '${rpc_config_file}' is prohibited when booted into 'PERSISTENT Mode - SYSMAINT Session'.

Will be notified using `notify-send` (passive popup) and `logger` (journal log), if available.

Applicable to all Qubes qrexec services listed in file `/usr/share/user-sysmaint-split/qubes-rpc-blocks`.

**Comment by ben-grande** [comment]
This issue was first closed as not planned because of EOL, nobody reported this issue here since. It later applied to Whonix due to unprivileged sysmain permissions, which was fixed. Closing as solved.
